pkg/rbac/userData.go

```diff
@@ -153,9 +153,32 @@ func (user *UserDataCache) userHasAllAccess(ctx context.Context, cache *Cache) (
 
 		cache.shared.mcCache.lock.Lock()
 		defer cache.shared.mcCache.lock.Unlock()
-		user.ManagedClusters = cache.shared.managedClusters
+		user.ManagedClusters = map[string]struct{}{"*": {}}
 		user.clustersCache.updatedAt = time.Now()
 		user.csrCache.err, user.nsrCache.err, user.clustersCache.err = nil, nil, nil
+		klog.V(5).Infof("User %s with uid %s has access to all resources.",
+			user.userInfo.Username, user.userInfo.UID)
+
+		return true, nil
+	} else if user.userAuthorizedListSSAR(ctx, impersClientSet,
+		"get", "searches", "searches/allManagedData") {
```

Comment:
**Future optimization**
This if/else requires 2 client requests which are executed synchronously. We could save a few milliseconds if we execute these asynchronously.